src/plenigo/internal/utils/JWT.php

```diff
@@ -0,0 +1,193 @@
+<?php
+
+namespace plenigo\internal\utils;
+
+/**
+ * <p>
+ * This class implements the generation of JWT Tokens
+ * </p>
+ * <p>
+ * <b>IMPORTANT:</b> This class is part of the internal API, please do not use it, because it can
+ * be removed in future versions of the SDK or access to such elements could
+ * be changed from 'public' to 'protected' or less.
+ * </p>
+ *
+ * @category SDK
+ * @package  PlenigoInternal
+ * @author   Sebastian Dieguez <s.dieguez@plenigo.com>
+ * @link     https://www.plenigo.com
+ *
+ */
+class JWT {
+
+    /**
+     * Decodes a JWT string into a PHP object.
+     *
+     * @param string      $jwt    The JWT
+     * @param string|null $key    The secret key
+     * @param bool        $verify Don't skip verification process 
+     *
+     * @return object      The JWT's payload as a PHP object
+     * @throws \Exception          Algorithm was not provided or the provided JWT was invalid
+     * 
+     * @uses jsonDecode
+     * @uses urlsafeB64Decode
+     */
+    public static function decode($jwt, $key = null, $verify = true) {
+        $tks = explode('.', $jwt);
+        if (count($tks) != 3) {
```

Comment:
a JWT token is a string consisting of 3 parts separated but dots

http://jwt.io